src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionStringBuilder.cs

```diff
@@ -646,6 +652,21 @@ public int MinPoolSize
             }
         }
 
+        /// <include file='..\..\..\..\..\..\..\doc\snippets\Microsoft.Data.SqlClient\SqlConnectionStringBuilder.xml' path='docs/members[@name="SqlConnectionStringBuilder"]/PoolIdleTimeout/*' />
+        public int PoolIdleTimeout
+        {
+            get { return _poolIdleTimeout; }
+            set
+            {
+                if (value < 0)
```

Comment:
This check for <0 will mean that users can't set it to the default value of -1 to return to the default behaviour which seems counter intuitive, both removing and setting to default value should normally yield the same result I think.

Reply:
We're only accepting values **0 to Int32.Max** as documented [here](https://github.com/dotnet/SqlClient/pull/348/files#diff-68c5ce9ab44eb1bae56a0568df742007R597).

This customized behaviour is disabled by default (hence -1) so setting to anything other than the range is essentially not allowed.

Same is with `Connection Timeout` although it's default is 15 secs. We not really obstructing default value, but anything beyond range. I was also thinking of whether I should also check for Max value here, but didn't see that at other places, but maybe we should check everywhere?

Reply:
> This customized behaviour is disabled by default (hence -1) so setting to anything other than the range is essentially not allowed.

Sure, but if you enable it how do you then disable it without starting again with a new connection string builder? Or is that not a suggested use case?

Reply:
Yes you're right. I will change it to **< -1**.

Reply:
Is 0 a useful value to allow? I would expect that to instantly dispose of pools which is the same thing as just not using pooling.

Reply:
The behavior is same, yes. Allowing 0 or not, doesn't seem a problem to me.

@David-Engel if you could confirm the range as well, I can update code accordingly.